In [1]:
from time import sleep
import threading

def sing():
    for i in range(2):
        print("sing ...")
        sleep(1)

def dance():
    for i in range(2):
        print("dancing ...")
        sleep(1)

t1 = threading.Thread(target=sing)
t2 = threading.Thread(target=dance)
t1.start()
t2.start()

sing ...
dancing ...


dancing ...
sing ...


In [4]:
import queue

q = queue.PriorityQueue()
for i in range(5):
    q.put((i, str(i+10)))

print(q.get())
print(q.get())
print(q.get())

(0, '10')
(1, '11')
(2, '12')


In [8]:
from pathlib import Path

root_train = r"D:\workspace\MYData\比赛\眼科比赛\eye_com\Train\Train"
train_num = 0
for sub_dir in Path(root_train).iterdir():
    train_num += len([p for p in sub_dir.glob("*.jpg")])
print(train_num)

root_val = r"D:\workspace\MYData\比赛\眼科比赛\eye_com\Validation\Validation_images"
val_num = 0
for sub_dir in Path(root_val).iterdir():
    val_num += len([p for p in sub_dir.glob("*.jpg")])
print(val_num)

33559
11110


训练集：33559张图片
验证集：11110张图片
在这些图片基础上，包含了14个多分类任务，每个任务的类别数量从 2 - 23 不等

In [10]:
pred_root = Path("D:\workspace\MYData\比赛\厨房\初赛试题\picture").glob("*.jpg")
pred_root = [i for i in pred_root]
print(len(pred_root))

1727


明厨亮灶：

一共有8个类别（初赛5个，决赛增加了3个）每个类别提供了 14 张训练图片
需要预测的图片数量: 1727
需要预测的视频数量：251个，每个视频时长在5秒左右，视频帧率为15或20帧（即1秒有15或20张图片）